model/project_aliases.go

```diff
@@ -200,19 +200,35 @@ func FindAliasInProjectOrRepo(projectID, alias string) ([]ProjectAlias, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "error checking for existing aliases")
 	}
-	// If nothing is defined in the DB, check the project config
-	// Unless the alias defined is a patch alias and there are patch aliases
+	// If nothing is defined in the DB, check the project config,
+	// unless the alias defined is a patch alias and there are patch aliases
 	// defined on the project page.
 	if len(aliases) > 0 {
 		return aliases, nil
 	}
 	shouldExit := false
 	if IsPatchAlias(alias) {
-		numPatchAliases, err := countPatchAliasesForProjectRef(projectID)
+		projectRefPatchAliases, err := countPatchAliases(projectID)
 		if err != nil {
 			return nil, errors.Wrap(err, "error counting patch aliases")
 		}
-		shouldExit = numPatchAliases > 0
+		shouldExit = projectRefPatchAliases > 0
+		if !shouldExit {
```

Comment:
We could always make a new ticket to improve this if that's easier, but this block of code is pretty repetitive with FindAliasInProjectOrRepoFromDb; it would be better if we could find a way for that function to give us more information. For example, it already does this count https://github.com/evergreen-ci/evergreen/blob/ffde3b1d0dc6e70aeea094410c86a288f50e35d9/model/project_aliases.go#L132 and decides whether or not we need to look at the repo. Perhaps the repo function https://github.com/evergreen-ci/evergreen/blob/ffde3b1d0dc6e70aeea094410c86a288f50e35d9/model/project_aliases.go#L248 should also now count in the case that there's no match, and then return whether or not we should exit (bc repo patch aliases exist) or continue onto the project config. That would save us two repeated DB calls in this function.

Reply:
Sounds reasonable - refactored.